### PR TITLE
MIAMI details

### PR DESCRIPTION
**Pull request by cloveras**
See https://github.com/vippsas/vipps-partner/pull/62

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,3 +136,28 @@ Vipps-System-Plugin-Version: 4.5.6
...
+[Management API](https://developer.vippsmobilepay.com/docs/APIs/management-api/)
+and
+[Report API](https://developer.vippsmobilepay.com/docs/APIs/report-api/)
+use a _new_ token endpoint:
+`POST:/miami/v1/token`.
+
+**Important:* This endpoint will be renamed to
+`POST:/authentication/v1/token`
+later, when the internal technical dependencies are resolved.
+
```
Comment (Haldormyre) [inline README.md:151]:
```suggestion
Authenticating with this endpoint is quite similar to the above mentioned flow, but this new endpoints uses a completely standard OAuth client credentials flow, allowing use of standardized libraries. We highly recommend this approach, using one of the [trusted](https://oauth.net/code/) libraries to perform the flow.
```

File: docs/swagger.yaml [hunk 2]
```diff
@@ -72,10 +83,63 @@ paths:
         "401":
           description: Unauthorized
         "500":
-          description: A server error on Vipps' side.
+          description: Server error.
+
+  /authentication/v1/token:
```
Comment (Haldormyre) [inline docs/swagger.yaml:88]:
```suggestion
  /miami/v1/token:
```

File: docs/swagger.yaml [hunk 3]
```diff
@@ -72,10 +83,63 @@ paths:
...
+      operationId: fetchToken
+      description: |-
+        This is the new authentication endpoint, based on OAuth (Open Authorization).
+        Requires only `client_id` and `client_secret`.
+        The `Ocp-Apim-Subscription-Key` is not used.
+      parameters:
+        - name: client_id
+          in: header
+          description: |-
+            The `client_id` is available on portal.vipps.no, under the 'Utvikler' section.
```
Comment (Haldormyre) [inline docs/swagger.yaml:101]:
```suggestion
            The `client_id` is available on portal.vipps.no, under the 'Developer' section.
```

File: docs/swagger.yaml [hunk 4]
```diff
@@ -72,10 +83,63 @@ paths:
...
+            Think of it as the "username".
+          required: true
+          schema:
+            type: string
+            format: guid
+          example: fb492b5e-7907-4d83-ba20-c7fb60ca35de
+        - name: client_secret
+          in: header
+          description: |-
+            The `client_secret` is available on portal.vipps.no, under the 'Utvikler' section.
```
Comment (Haldormyre) [inline docs/swagger.yaml:111]:
```suggestion
            The `client_secret` is available on portal.vipps.no, under the 'Developer' section.
```

File: README.md [hunk 5]
```diff
@@ -20,6 +20,13 @@ END_METADATA -->
 [Vipps MobilePay Technical Documentation](https://developer.vippsmobilepay.com/docs/APIs/access-token-api/).
 <!-- END_COMMENT -->
 
+**Important:** Some of the new APIs, including the
+[Management API](https://developer.vippsmobilepay.com/docs/APIs/management-api/)
+and
+[Report API](https://developer.vippsmobilepay.com/docs/APIs/report-api/)
+use a new token endpoint. See
```
Comment (okpedersen) [inline README.md:27]: Nitpicking: You don't need to use the new token endpoint, depending on your role. I think the current phrasing might lead to questions, to me it implies that everyone must use new endpoint to get access :thinking:

Management API: Management partners use new endpoint, partners & merchants use old endpoint.

Report API: Accounting partners use new endpoint, merchants use old, partners does not have access

File: README.md [hunk 6]
```diff
@@ -129,3 +136,28 @@ Vipps-System-Plugin-Version: 4.5.6
 `HTTP 401 Unauthorized` error. See the FAQ:
 [Why do I get `HTTP 401 Unauthorized?`](https://developer.vippsmobilepay.com/docs/knowledge-base/errors#why-do-i-get-http-401-unauthorized).
 
+## Token endpoint
+
+Some of the new APIs, including the
+[Management API](https://developer.vippsmobilepay.com/docs/APIs/management-api/)
+and
+[Report API](https://developer.vippsmobilepay.com/docs/APIs/report-api/)
+use a _new_ token endpoint:
```
Comment (okpedersen) [inline README.md:145]: Same as above

File: README.md [hunk 7]
```diff
@@ -129,3 +136,28 @@ Vipps-System-Plugin-Version: 4.5.6
...
+`POST:/miami/v1/token`.
+
+**Important:* This endpoint will be renamed to
+`POST:/authentication/v1/token`
+later, when the internal technical dependencies are resolved.
+Authenticating with this endpoint is quite similar to the above mentioned flow, but this new endpoints uses a completely standard OAuth client credentials flow, allowing use of standardized libraries. We highly recommend this approach, using one of the [trusted](https://oauth.net/code/) libraries to perform the flow.
+Example request to `POST:/miami/v1/token`:
```
Comment (okpedersen) [inline README.md:152]: I will create some examples

File: docs/swagger.yaml [hunk 8]
```diff
@@ -72,10 +83,63 @@ paths:
...
+          description: Bad request
+        "401":
+          description: Unauthorized
+        "500":
+          description: Server error.
 
 components:
   schemas:
+    TokenResponse:
+      type: object
```
Comment (okpedersen) [inline docs/swagger.yaml:141]: We should improve this, will create something

**Pull request merged**